core/processor.go

```diff
@@ -0,0 +1,544 @@
+		return fmt.Errorf("invalid botDelegatedWithdrawToEvent: %v", botDelegatedWithdrawToEvent)
+	}
+
+	messagePassedLog := receipt.Logs[botDelegatedWithdrawToEvent.LogIndex-3]
+	sentMessageLog := receipt.Logs[botDelegatedWithdrawToEvent.LogIndex-2]
+	sentMessageExtension1Log := receipt.Logs[botDelegatedWithdrawToEvent.LogIndex-1]
+
+	sentMessageEvent, err := b.toL2CrossDomainMessengerSentMessageExtension1(sentMessageLog, sentMessageExtension1Log)
+	if err != nil {
+		return err
+	}
+	messagePassedEvent, err := b.toMessagePassed(messagePassedLog)
+	if err != nil {
+		return err
+	}
+
+	withdrawalTx, err := b.toLowLevelMessage(sentMessageEvent, messagePassedEvent)
+	if err != nil {
+		return fmt.Errorf("toLowLevelMessage err: %v", err)
+	}
+
+	hash, err := b.hashWithdrawal(withdrawalTx)
+	if err != nil {
+		return fmt.Errorf("hashWithdrawal err: %v", err)
+	}
+
+	messageSlot, err := b.hashMessageHash(hash)
+	if err != nil {
+		return fmt.Errorf("hashMesaageHash err: %v", err)
+	}
+
+	l2OutputIndex, outputProposal, err := b.getL2OutputAfter(l2BlockNumber)
+	if err != nil {
+		return err
+	}
+	accountResult, err := b.L2Client.GetProof(
+		b.L2Contracts.L2ToL1MessagePasser,
+		[]string{"0x" + messageSlot},
+		outputProposal.L2BlockNumber,
+	)
+	if err != nil {
+		return fmt.Errorf("GetProof err: %v", err)
+	}
+
+	outputProposalBlock, err := b.L2Client.HeaderByNumber(context.Background(), outputProposal.L2BlockNumber)
+	if err != nil {
+		return fmt.Errorf("get output proposal block error: %v", err)
+	}
+
+	withdrawalProof := accountResult.StorageProof[0]
+	withdrawalProof2Bytes := make([][]byte, 0)
+	for _, p1 := range withdrawalProof.Proof {
+		withdrawalProof2Bytes = append(withdrawalProof2Bytes, p1)
+	}
+
+	outputRootProof := bindings.TypesOutputRootProof{
+		Version:                  common.HexToHash("0x"),
+		StateRoot:                outputProposalBlock.Root,
+		MessagePasserStorageRoot: accountResult.StorageHash,
+		LatestBlockhash:          outputProposalBlock.Hash(),
+	}
+
+	l1ChainId, err := b.L1Client.ChainID(context.Background())
+	if err != nil {
+		return err
+	}
+
+	// Retrieve the private key from the environment variable OPBNB_BRIDGE_BOT_PRIVKEY
+	botPrivkey, err := crypto.HexToECDSA(os.Getenv("OPBNB_BRIDGE_BOT_PRIVKEY"))
+	if err != nil {
+		return fmt.Errorf("invalid $OPBNB_BRIDGE_BOT_PRIVKEY: %w", err)
+	}
+
+	pubKey := botPrivkey.Public()
+	pubKeyECDSA, ok := pubKey.(*ecdsa.PublicKey)
+	if !ok {
+		return errors.New("cannot assert type: publicKey is not of type *ecdsa.PublicKey")
+	}
+	pubKeyBytes := crypto.FromECDSAPub(pubKeyECDSA)
+	pubKeyHash := crypto.Keccak256(pubKeyBytes[1:])[12:]
+	fromAddress := common.HexToAddress(hexutil.Encode(pubKeyHash))
+
+	const gasPrice = 9000000000 // 9 GWei
+
+	optimismPortalTransactor, _ := bindings.NewOptimismPortalTransactor(
+		b.L1Contracts.OptimismPortalProxy,
+		b.L1Client,
+	)
+	signedTx, err := optimismPortalTransactor.ProveWithdrawalTransaction(
```

Comment:
will it dry run before send?
what if the tx fails?
what if the tx pending for a long time?

Reply:
1. It will estimate gas before sending. Eth_estimateGas can be considered a dry run.

2. No retry for tx failures. These unproven withdrawals will be rescanned by the processor goroutine and retried to prove.

3. As (2) says, I think failed transactions will be retried by the processor goroutine.

I am finding a way to enhance the transaction management, please touch me if you have some ideas.